packages/airnode-node/src/evm/gas-prices.ts

```diff
@@ -1,47 +1,53 @@
 import { BigNumber, ethers } from 'ethers';
+import { parseEther } from 'ethers/lib/utils';
 import { go } from '../utils/promise-utils';
 import * as logger from '../logger';
-import { BASE_FEE_MULTIPLIER, DEFAULT_RETRY_TIMEOUT_MS, PRIORITY_FEE } from '../constants';
-import { GasTarget, LogsData, PendingLog } from '../types';
+import { BASE_FEE_MULTIPLIER, DEFAULT_RETRY_TIMEOUT_MS, PRIORITY_FEE, WEI_PER_GWEI } from '../constants';
+import { ChainOptions, GasTarget, LogsData } from '../types';
 
 interface FetchOptions {
   readonly provider: ethers.providers.JsonRpcProvider;
+  readonly chainOptions: ChainOptions;
 }
 
 export const getGasPrice = async (options: FetchOptions): Promise<LogsData<GasTarget | null>> => {
-  const { provider } = options;
-  const [logs, blockGas] = await (async (): Promise<[PendingLog[], GasTarget | null]> => {
-    const operation = () => provider.getBlock('latest');
-    const [err, blockHeader] = await go(operation, { retries: 1, timeoutMs: DEFAULT_RETRY_TIMEOUT_MS });
-    if (err || !blockHeader?.baseFeePerGas) {
-      const log = logger.pend('INFO', 'Failed to get EIP-1559 gas pricing from provider - trying fallback', err);
-
-      return [[log], null];
+  const { provider, chainOptions } = options;
+
+  switch (chainOptions.txType) {
+    case '1': {
+      const operation = () => provider.getGasPrice();
+      const [err, gasPrice] = await go(operation, { retries: 1, timeoutMs: DEFAULT_RETRY_TIMEOUT_MS });
+      if (err || !gasPrice) {
+        const log = logger.pend('ERROR', 'Failed to get legacy gas price from provider', err);
+        return [[log], null];
+      }
+
+      return [[], { gasPrice }];
     }
-
-    const maxPriorityFeePerGas = BigNumber.from(PRIORITY_FEE);
-    const maxFeePerGas = blockHeader.baseFeePerGas.mul(BASE_FEE_MULTIPLIER).add(maxPriorityFeePerGas);
-
-    return [
-      [],
-      {
-        maxPriorityFeePerGas,
-        maxFeePerGas,
-      } as GasTarget,
-    ];
-  })();
-
-  if (blockGas) {
-    return [logs, blockGas];
-  }
-
-  // Fallback to pre-EIP-1559
-  const operation = () => provider.getGasPrice();
-  const [err, gasPrice] = await go(operation, { retries: 1, timeoutMs: DEFAULT_RETRY_TIMEOUT_MS });
-  if (err || !gasPrice) {
-    const log = logger.pend('ERROR', 'Failed to get fallback gas price from provider', err);
-    return [[...logs, log], null];
+    case '2': {
+      const operation = () => provider.getBlock('latest');
```

Comment:
Do we maybe want to extract each case into smaller, descriptive functions?

Reply:
Re: `parseEther`: this is about to be invalidated due to implementation of Burak's suggestion around a config object for `priorityFee` that will have value and unit fields (which underneath with use a different function).

Re: the cases as separate functions: I'll do that now.

Reply:
Done :)